feeds/wifi-ax/mac80211/patches/qca/226-ath11k-set-registered-flag-before-qdss-config.patch

```diff
@@ -0,0 +1,11 @@
+diff -Naur a/drivers/net/wireless/ath/ath11k/core.c b/drivers/net/wireless/ath/ath11k/core.c
+--- a/drivers/net/wireless/ath/ath11k/core.c	2021-09-30 09:44:22.861587609 -0400
++++ b/drivers/net/wireless/ath/ath11k/core.c	2021-09-30 09:45:18.361000898 -0400
+@@ -847,6 +847,7 @@
+ #endif
+ 	ath11k_hif_irq_enable(ab);
+ 
++	set_bit(ATH11K_FLAG_REGISTERED, &ab->dev_flags);
```

Comment:
the ath11k_core_qmi_firmware_ready is also called form ath11k_core_reconfigure_on_crash, and in that function the ATH11K_FLAG_REGISTERED is not set. So this flag needs to be set only when the ATH11K_QMI_EVENT_FW_READY event comes and not in the other case. 

Also, is it possible to set the number of attempts it tries to get the qdss fw, so we can limit the number of tries to 1 and it wouldn't cause much delay.

Reply:
ATH11K_FLAG_REGISTERED is necessary for the Country code setting. Setting it in ath11k_core_qmi_firmware_ready() from ath11k_core_reconfigure_on_crash() does not seemed to cause any problem, and in-fact seems necessary.

Reply:
Regarding the retries comment. There does not seemed to be multiple retries. Once it tries from Kernel which if fails, it takes User's help to load the file. In both the modes, it is tried only once.